listenbrainz/webserver/views/api_tools.py

```diff
@@ -153,18 +153,24 @@ def validate_listen(listen, listen_type):
                                 "key when submitting playing_now.", listen)
 
     # Basic metadata
-    try:
-        if not listen['track_metadata']['track_name']:
-            raise APIBadRequest("JSON document does not contain required "
-                                "track_metadata.track_name.", listen)
-        if not listen['track_metadata']['artist_name']:
-            raise APIBadRequest("JSON document does not contain required "
-                                "track_metadata.artist_name.", listen)
+    if 'track_name' in listen['track_metadata']:
+        if len(listen['track_metadata']['track_name']) == 0:
+            raise APIBadRequest("required field track_metadata.track_name is empty.", listen)
+        listen['track_metadata']['track_name'] = listen['track_metadata']['track_name'].strip()
+    else:
+        raise APIBadRequest("JSON document does not contain required track_metadata.track_name.", listen)
+
+
+    if 'artist_name' in listen['track_metadata']:
         if not isinstance(listen['track_metadata']['artist_name'], str):
             raise APIBadRequest("artist_name must be a single string.", listen)
-    except KeyError:
-        raise APIBadRequest("JSON document does not contain a valid metadata.track_name "
-                       "and/or track_metadata.artist_name.", listen)
+
+        if len(listen['track_metadata']['artist_name']) == 0:
+            raise APIBadRequest("required field track_metadata.artist_name is empty.", listen)
+        listen['track_metadata']['artist_name'] = listen['track_metadata']['artist_name'].strip()
```

Comment:
Should we call strip *before* checking its length for the artist name?
Otherwise an artist name with only whitespace characters will be stripped to an empty string.
I thought you mentioned the artist name should never be empty.

Reply:
Good catch. Move the tests to after the strip. As far as artist names being empty, I'm now confused as to what it means for an artist to be empty. lol. I think the code is fine as it is now.